src/network/connection.js

```diff
@@ -371,7 +376,18 @@ module.exports = class Connection {
       return this.authHandlers.onSuccess(rawData)
     }
 
-    this.buffer = Buffer.concat([this.buffer, rawData])
+    this.bufferQueue.buffers.push(rawData)
+    this.bufferQueue.bytesTotal += Buffer.byteLength(rawData)
+
+    const bytesTotal = Buffer.byteLength(this.buffer) + this.bufferQueue.bytesTotal
+    if (this.bufferQueue.bytesAwaiting <= bytesTotal) {
```

Comment:
Can you inverse this statement please. It would read a lot clearer if it's just "if there are more bytes to read, return" than the way it is now.

Reply:
Got it, I rewrote the patch to make it easier to follow